scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -487,6 +489,13 @@ def run_command(self, cmd, timeout):
                 self.state = 'failed'
                 proc.kill()
 
+        if proc.returncode in (ExitCode.INTERRUPTED, ExitCode.USAGE_ERROR, ExitCode.INTERNAL_ERROR):
+            self.state = 'failed'
+            self.instance.reason = f'Pytest failed with return code {proc.returncode}'
+            with open(self.pytest_log_file_path, 'w') as log_file:
```

Comment:
```suggestion
            with open(self.pytest_log_file_path, 'a') as log_file:
```

Reply:
I intentionally used `write` mode instead of `append`. If we receive an error from pytest, then there is no log file. This log file can exist when the test is re-run, but we do not want to include logs from the previous test execution.